proxy/src/main.rs

```diff
@@ -22,31 +22,29 @@ use url::Url;
 const ENV_CONFIG: &str = "SD_PROXY_ORIGIN";
 
 #[cfg(feature = "qubesdb")]
-fn read(name: &str) -> Option<String> {
-    let mut path = String::from("/vm-config/");
-    path.push_str(name);
-
-    let _value = unsafe {
-        let _path = CString::new(path).unwrap().into_raw();
-        let mut len: u32 = 0;
+fn read(name: &str) -> Result<String> {
+    let path = format!("/vm-config/{name}");
 
+    let path_raw = CString::new(path.clone()).unwrap().into_raw();
+    let mut len: u32 = 0;
+    let value_raw = unsafe {
         let db = qdb_open(ptr::null_mut());
-        let value = qdb_read(db, _path, &mut len);
+        let value_unsafe = qdb_read(db, path_raw, &mut len);
         qdb_close(db);
+        let _ = CString::from_raw(path_raw); // retake
 
-        let _ = CString::from_raw(_path);
         if len > 0 {
             CStr::from_ptr(value_unsafe)
         } else {
             bail!("Could not read from QubesDB: {}", path);
         }
     };
-    let value = _value
+
+    let value = value_raw
         .to_owned()
         .into_string()
         .expect("QubesDB value should be a valid string");
```

Comment:
This can be a ? instead of .expect now

Reply:
Thanks; fixed since 238a3f5f71b620b02785d34e9e8742785be1c1fc.